app/MapScreen.js

```diff
@@ -24,33 +51,140 @@ const styles = StyleSheet.create({
     flex: 1,
     width: '100%',
   },
+  header: {
+    backgroundColor: transparentBlack,
+    padding: 14,
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    width: screenWidth,
+  },
+  headerText: {
+    color: colorWhite,
+    fontSize: 14,
+  },
+  microphoneButton: {
+    position: 'absolute',
+    bottom: 40,
+    right: screenWidth / 2 - 40,
+  },
+  buttonContainer: {
+    backgroundColor: colorWhite,
+    borderRadius: 64,
+    width: 80,
+    height: 80,
+    alignItems: 'center',
+    elevation: 1,
+    padding: 1,
+    paddingTop: 10,
+    shadowOffset: { width: 0, height: 0 },
+    shadowOpacity: 1,
+  },
+  errorView: {
+    backgroundColor: transparentBlack,
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    width: screenWidth,
+    height: screenHeight,
+    padding: 50,
+    paddingTop: 80,
+  },
+  errorText: {
+    color: colorWhite,
+    fontSize: 26,
+  },
 });
 
 export default class MapScreen extends Component {
   constructor(props) {
     super(props);
 
     const { navigation } = this.props;
-    const centerCoords = navigation.getParam('centerCoords', null);
-    const lexSlotValues = navigation.getParam('lexSlotValues', {});
+    const centerCoords = navigation.getParam('centerCoords', [
+      -77.0368707,
+      38.9071923,
+    ]);
+
+    const tileQueryParamsUI = {
+      bandCombination: bandCombinationLabels.natural,
+    };
 
     this.state = {
       centerCoords,
-      lexSlotValues,
-      tileQueryString: '',
+      tileQueryParamsUI,
+      tileQueryString: null,
+      isAuthorized: false,
+      isRecording: false,
+      animatedShadowRadius: new Animated.Value(inactiveShadowRadius),
+      errorMessage: null,
     };
   }
 
   componentDidMount() {
     this.convertLexSlotsToQueryParams();
+
+    AudioRecorder.requestAuthorization().then((isAuthorized) => {
+      this.setState({ isAuthorized });
+    });
+
+    this.prepareRecordingAnimation();
+    this.prepareRecorder();
+  }
+
+  setErrorMessage(message) {
+    this.setState({
+      errorMessage: message,
+    });
+  }
+
+  onAudioRecordingFinished = async (data) => {
+    // Android callback comes in the form of a promise instead.
+    if (Platform.OS === 'ios') {
+      this.finishRecording(data.audioFileURL);
+    }
+
+    const { isRecording } = this.state;
+
+    if (isRecording) {
+      return;
+    }
+
+    const errorMessage = 'Sorry, we didn\'t understand that. Please try again';
+
+    if (!data.base64) {
+      this.setState({
```

Comment:
Should this be a `setErrorMessage` call?

Reply:
Yeah, I'll update that.